pkg/controllers/cluster/resource/resource.go

```diff
@@ -74,6 +82,34 @@ func MemberServiceForPod(pod *corev1.Pod, cluster *scyllav1.ScyllaCluster) *core
 			PublishNotReadyAddresses: true,
 		},
 	}
+
+	if cluster.Spec.Network.HostNetworking {
+		service.Spec.ClusterIP = corev1.ClusterIPNone
```

Comment:
It is more to avoid confusion of IP used and to ensure no bug in some release that would leads to use that ClusterIP with host network. As it will be None it will failed and will be easy to identify the issue while with an IP we could start use that IP...